### PR TITLE
FOB Control Points

### PR DESCRIPTION
**Pull request by walterroach**
Initial implementation of FOBs as connected control points.

A FOB can be defined in a campaign .miz by placing a CJTF Blue CP_SKP_11_ATC_Mobile_Command_Post.

The group name of the unit is used to name the FOB.  

Each FOB gets the existing static template for FOBs (There's only one currently, we should add more and confirm compatibility with various factions).  It also gets defenses the same way airbases do, except no medium/long range SAMs.  

#385

### CODE REVIEW

**Comment by walterroach** [comment]
FOBs aren't capturable except via cheat in this PR

**Comment by DanAlbert** [comment]
Allowing medium range SAMs might be good. The mobile SAMs seem like they belong there and they currently fit into that group. An SA-3 or Hawk also make sense.

**Comment by Khopa** [comment]
I checked out the branch, but i don't have any FOB in the Inherent Resolve campaign. Maybe the merge might have overriden your changes in the miz file ?

Edit : (I have the FOB appearing if i checkout the revisions before the merge)

**Comment by walterroach** [comment]
> I checked out the branch, but i don't have any FOB in the Inherent Resolve campaign. Maybe the merge might have overriden your changes in the miz file ?
> 
> Edit : (I have the FOB appearing if i checkout the revisions before the merge)

I intentionally reverted the campaign change since the FOB isn't able to be captured yet (Also, its location was probably not very useful).

File: game/theater/controlpoint.py [hunk 1]
```diff
@@ -623,3 +623,56 @@ def active_runway(self, conditions: Conditions,
...
+        return False
+
+    @property
+    def heading(self) -> int:
+        return 0
+
+    # def active_runway(self, conditions: Conditions,
+    #                   dynamic_runways: Dict[str, RunwayData]) -> RunwayData:
+    #     logging.warning("TODO: FOBs have no runways")
+    #     return RunwayData(self.full_name, runway_heading=0, runway_name="")
```
Comment (DanAlbert) [inline game/theater/controlpoint.py]: Think you probably meant to remove this :)

File: game/theater/start_generator.py [hunk 2]
```diff
@@ -537,7 +538,27 @@ def generate_shorad(self) -> None:
             return
         g.groups.append(group)
         self.control_point.base_defenses.append(g)
-
+class FobDefenseGenerator(BaseDefenseGenerator):
+    def generate(self) -> None:
+        self.generate_garrison()
+        self.generate_fob_defenses()
+        # self.generate_fob()
```
Comment (DanAlbert) [inline game/theater/start_generator.py]: And here

**Review by DanAlbert — APPROVED** [review]

**Pull request merged**